wsgi_kerberos.py

```diff
@@ -66,23 +66,15 @@ class KerberosAuthMiddleware(object):
     '''
     WSGI Middleware providing Kerberos Authentication
 
-    If no hostname is provided, the name returned by socket.gethostname() will
-    be used.
-
     :param app: WSGI Application
-    :param hostname: Hostname for kerberos name canonicalization
-    :type hostname: str
     :param unauthorized: 401 Response text or text/content-type tuple
     :type unauthorized: str or tuple
     :param forbidden: 403 Response text or text/content-type tuple
     :type forbidden: str or tuple
     '''
 
-    def __init__(self, app, hostname=None, unauthorized=None, forbidden=None,
+    def __init__(self, app, unauthorized=None, forbidden=None,
```

Comment:
I don’t think this is a 100% good change. I think the doc change to not push people to setting the hostname is good, and a change in behavior when it’s not provided (still None) is warranted to make the default to not restrict things, but dropping the option to restrict it isn’t something I’d support.

Reply:
If you really want to allow people that level of low-level control, I think it'd be better to provide a service_name param instead of a hostname, to allow full customization of the service name to be used, and to discourage existing uses of the hostname param from being kept.

Reply:
And by service name, you mean allow them to override the `HTTP` part of the gss service name? This is for wsgi services, so I don’t see that as being all that beneficial, as wsgi pretty much implies http.

I also don’t think that or the existing proposed change are sufficient to warrant a change in API for users to deal with.

Reply:
@mkomitee, I have reworked this changeset in response to your concerns. I no longer drop the `hostname` parameter from the constructor, allowing users who do pass it to still see a warning if no corresponding server principal is found, but preserve the change to pass `''` rather than the result of `socket.gethostname()` to the `kerberos.authGSSServerInit()` call, which is the crux of the fix for #3. Please let me know if this looks better. Thanks!